### ISSUE TITLE
[Roadmap] Reward / penalty learning research

### ISSUE THREAD

**Issue by QWQOUOT485** [QWQOUOT485/siri#66]
## Goal

Track a future **reward / penalty learning** research direction for the Siri Agent.

This is roadmap-only. It does **not** authorize training, fine-tuning, online learning, production Local AI fallback, Semantic Memory enablement, or any expansion of execution authority.

## Core idea

Future experiments may use reward-based optimization or preference / reinforcement-style training to improve bounded semantic behavior.

- **Reward**: positive training signal for desirable behavior.
- **Penalty / negative reward**: negative training signal for undesirable behavior.
- Some penalties may be implemented as negative reward; others may remain separate costs or regularization terms.

Conceptually:

```text
correct intent / grounded spans        -> positive reward
correct abstention on unknown          -> positive reward
ordinary semantic miss                 -> moderate penalty
false acceptance / unsafe guess        -> large penalty
policy / authority boundary violation  -> hard rejection + large penalty
```

The hard rejection occurs outside the model and can never be overridden by aggregate reward.

## Candidate positive signals

Future reward design may consider positive signal for:

- correct bounded intent;
- correct `track` / `artist` / `album` spans;
- correct `unknown` / abstention;
- strict-schema output;
- valid grounding through the existing deterministic resolver;
- calibrated uncertainty when the model should defer.

## Candidate penalties

Potential negative signal may include:

- false acceptance;
- predicting `spotify_play_track` when the correct result is `unknown`;
- malformed structured output;
- ungrounded entity spans;
- unsupported claims;
- deterministic-only or safety-only requests reaching the model unexpectedly;
- attempts to produce or bypass execution authority.

Safety mistakes should carry a much larger penalty than ordinary semantic misses.

Over-abstention may receive only a small cost, and only after safety recall is protected.

Latency / memory may be secondary costs, but may never be traded against safety.

## Reward-hacking guardrails

Real execution success must **not** be the sole reward signal.

In particular:

- Spotify playback success must not make a model-produced Spotify ID/URI trusted.
- Process execution success must not legitimize shell/path authority.
- No URL/path/process/provider target may become trusted because it earns reward.
- A policy bypass can never be compensated by a higher total score.
- Hard security constraints remain deterministic constraints, not soft reward terms.

The model may optimize only bounded semantic evidence:

```text
model semantic evidence
-> deterministic grounding
-> policy
-> closed ValidatedAction
-> trusted adapter
```

## Data discipline

Future reward / preference evidence must preserve current evaluation boundaries:

- frozen Stage A remains test-only;
- sealed Stage B held-out data must not tune reward weights, penalties, thresholds, or reward models;
- raw private logs must not be ingested automatically;
- human-confirmed corrections may become sanitized candidate preference/reward evidence only after review;
- no online self-training from live production actions is authorized.

## Future experiment protocol

Any future RL / preference / reward-model experiment should require a separately frozen and independently reviewed protocol covering at least:

1. model/base/adapter identity;
2. training data identity and hashes;
3. reward specification and weights;
4. penalty specification and weights;
5. hard safety constraints that cannot be traded off;
6. train/validation/held-out separation;
7. reward-hacking checks;
8. offline evaluation;
9. reproducibility and version/hash tracking;
10. independent review before escalation.

Potential research families may include RLHF, RLAIF, RLVR, or project-owned deterministic/verifiable reward signals, but none is selected or authorized by this issue.

## Relationship to current Stage B

This issue is future research only.

It must not modify or bypass the current Stage B sequence, hardware/model qualification gates, corpus freeze, held-out seal, or independent review flow.

Current authority state remains:

```text
training_authorized=false
model_compute_authorized=false
semantic_memory_enabled=false
local_ai_fallback_approved=false
```

## Non-actions

This issue does **not** authorize:

- training or fine-tuning;
- creating a reward model;
- running RLHF / RLAIF / RLVR;
- changing frozen Stage A or Stage B artifacts;
- using held-out data for reward tuning;
- online self-learning from user actions;
- enabling Semantic Memory;
- enabling Local AI fallback;
- changing Spotify / Windows / Siri execution authority;
- weakening deterministic grounding or policy;
- allowing learned reward to override hard security rules.

## Why keep this

Long term, reward + penalty learning may help the Agent learn a better tradeoff between:

- understanding the user correctly;
- extracting the right bounded entities;
- knowing when to abstain;
- avoiding false execution;
- keeping latency and resource use reasonable.

The central rule is that **semantic quality may be learned, but execution authority remains deterministic**.
